llvmDebugInfoC/src/scripts/konan_lldb.py

```diff
@@ -23,34 +23,48 @@
 
 import lldb
 import struct
+import os
+
+def is_appcode():
+    return 'AppCode' in os.getenv('XPC_SERVICE_NAME', '')
+
+def is_xcode():
+    return 'Xcode' in os.getenv('CLIENT_TYPE', '')
 
 NULL = 'null'
 
+CLIENT_APPCODE = is_appcode()
+CLIENT_XCODE = is_xcode()
+VISUAL_DEBUGGER = CLIENT_APPCODE or CLIENT_XCODE
+
 def log(msg):
     if False:
         print(msg)
 
+def exelog(stmt):
+    if False:
+        f = open(os.getenv('HOME', '') + "/lldbexelog.txt", "a")
+        f.write(stmt)
+        f.write("\n")
+        f.close()
+
 def lldb_val_to_ptr(lldb_val):
     addr = lldb_val.GetValueAsUnsigned()
     return '((struct ObjHeader *) {:#x})'.format(addr)
 
 
 def evaluate(expr):
     result = lldb.debugger.GetSelectedTarget().EvaluateExpression(expr, lldb.SBExpressionOptions())
-    log("{} => {}".format(expr, result))
+    evallog = "{} => {}".format(expr, result)
+    log(evallog)
+    exelog(evallog)
     return result
 
 def is_instance_of(addr, typeinfo):
     return evaluate("(bool)IsInstance({}, {})".format(addr, typeinfo)).GetValue() == "true"
 
-
-def is_string(value):
-    return is_instance_of(lldb_val_to_ptr(value), "theStringTypeInfo")
-
-
-def is_array(value):
-    return evaluate("(int)Konan_DebugIsArray({})".format(lldb_val_to_ptr(value))).unsigned == 1
-
+def is_string_or_array(value):
```

Comment:
Reduced evaluate calls. There are still several calls per-instance, and reducing them will improve performance.